### DISCUSSION TITLE
Welcome to Power-BI-Sidetools Discussions!

### DISCUSSION THREAD

**Discussion by didierterrien** [didierterrien/Power-BI-Sidetools#6] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  Please do not post issues here. Feel free to post them in the "Issues" tab.

**Comment by Josebismart** [comment]
Hi Didider,

Thanks a lot for such an amazing tool.

I am trying to use the idea of the DAX Parser visualization to generate a file where I could inspect the DAX measures in a very big model.

There is a way of using this tool for retrieve a tree for all the measures at once?

I was thinking in the idea of generating a json with such sctructure.

I really appreciate your work and attention.

Regards

**Reply by didierterrien** [reply]
Hello Jose,
Thanks for your interest!
How do you want to build this tool ? Which programming language, which environment ?

**Comment by Josebismart** [comment]
Hi Didier,

Thanks for your answer.

My objective is to build a solution to analyse the path of every single measure. I want to know if we have similar or duplicates measures.

To do so, would be very important to gather anyhow the tree information that your Parser does and so I could be able to compare those “calculation paths” with the rest of measures very easily.

But I have no deep knowledge of programming so I would love the possibility to extract the tree as a json that I could parse later. I use Power query for that… Yes I know… I suffer the pain of my misknowledge…

I would appreciate very much a bit of help here. I also think that this would add a lot of value to the tool as I didn’t found in internet a solution for similar or duplicates measures.

Thanks a lot!!

Regards

José Antonio García | Senior BI Consultant
***@***.***   ***@***.***  +34 931 786 880   ***@***.***  ***@***.******@***.***>   ***@***.*** @bismart2009<https://twitter.com/bismart2009>   ***@***.***  www.bismart.com<http://www.bismart.com/>
***@***.***<https://bismart.com/>
You can contact our Privacy Department in order to exercise your rights of access, rectification, cancellation, portability, limitation or opposition to the processing of data through the following e-mail address: ***@***.******@***.***> or by post addressed to the company name Bismart Business Intelligence Specialist Services SL. Castella 41-43, 08018 Barcelona. If you wish to stop receiving this communication, you can unsubscribe by sending an email to ***@***.******@***.***>. You can consult our Privacy Policy at any time.

De: Didier Terrien ***@***.***>
Enviado el: martes, 5 de septiembre de 2023 10:22
Para: didierterrien/Power-BI-Sidetools ***@***.***>
CC: José Antonio García ***@***.***>; Comment ***@***.***>
Asunto: Re: [didierterrien/Power-BI-Sidetools] Welcome to Power-BI-Sidetools Discussions! (Discussion #6)


Hello Jose,
Thanks for your interest!
How do you want to build this tool ? Which programming language, which environment ?

**Reply by didierterrien** [reply]
Hello Jose,
A good practice is not to create a new discussion to reply to a previous one.
I suggest you check Power BI Cleaner from Imke Feldmann within Power BI Sidetools (in the 'More' menu). I feel she provides the queries you need.
You can duplicate the Pbit file and modify queries. Then, you will be able to create an icon in External tools ribbon in order to launch your tool with one click
